lib/react_client.dart

```diff
@@ -614,11 +614,25 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
       });
 
   void handleComponentDidCatch(Component2 component, dynamic error, ReactErrorInfo info) => zone.run(() {
-        component.componentDidCatch(error, info);
+        // Due to the error object being passed in from ReactJS it is a javascript object that does not get dartified.
+        // To fix this we throw the error again from dart to the JS side and catch it dart side which re-dartifies it.
+        try {
+          throwErrorFromJS(error);
+        } catch (e, stack) {
+          // The dart stack track gets lost so we manually add it to the info object for referance.
+          info.dartStackTrace = stack;
```

Comment:
Can we add test coverage that this is populated properly?

Reply:
also i need to capitalize dart... apparently...